zcash_client_sqlite/src/wallet/init.rs

```diff
@@ -411,6 +411,103 @@ impl<P: consensus::Parameters> RusqliteMigration for WalletMigration2<P> {
     }
 }
 
+struct WalletMigrationAddTxViews;
+
+migration!(
+    WalletMigrationAddTxViews,
+    "282fad2e-8372-4ca0-8bed-71821320909f",
+    ["be57ef3b-388e-42ea-97e2-678dafcf9754"],
+    "Add views over transaction & note data."
+);
+
+impl RusqliteMigration for WalletMigrationAddTxViews {
+    type Error = WalletMigrationError;
+
+    fn up(&self, transaction: &Transaction) -> Result<(), WalletMigrationError> {
+        transaction.execute_batch(
+            "CREATE VIEW v_tx_sent AS
+            SELECT transactions.id_tx         AS id_tx,
+                   transactions.block         AS mined_height,
+                   transactions.tx_index      AS tx_index,
+                   transactions.txid          AS txid,
+                   transactions.expiry_height AS expiry_height,
+                   transactions.raw           AS raw,
+                   SUM(sent_notes.value)      AS sent_total,
+                   COUNT(sent_notes.id_note)  AS sent_note_count,
+                   SUM(CASE WHEN sent_notes.memo IS NULL OR sent_notes.memo = '' THEN 0 ELSE 1 END) AS memo_count,
+                   blocks.time                AS block_time
+            FROM   transactions
+                   JOIN sent_notes
+                          ON transactions.id_tx = sent_notes.tx
```

Comment:
I see `transactions.id_tx` has a unique constraint, which means the column is implicitly indexed.  But there's no unique constraint on `sent_notes.tx` (just a NOT NULL).  Should we add an index to `sent_notes.tx` to improve the performance of this join?

Reply:
Are columns that have FOREIGN KEY constraints on them not indexed in sqlite? If that's the case, then yes we should probably add an index; most of the time it shouldn't matter too much since for ordinary wallets this won't be huge numbers of notes.

Reply:
Would adding this index require modifying the view, or can the index be added independently? If the latter then we can leave it until we actually know what the performance of the view is like, but if we'd have to drop and recreate the view to use the index then I'd be more inclined to fold it into this migration.

Reply:
Foreign key wouldn't automatically cause sent_notes.tx to be indexed.  But my understanding is that a foreign key constraint in the sent_notes table would require transactions.id_tx to be indexed (e.g. with PRIMARY KEY, UNIQUE).